src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -169,14 +192,153 @@ export const selectRewardsByAccountSpecifier = createDeepEqualOutputSelector(
   }
 )
 
-export const selectRewardsAmountByDenom = createSelector(
+export const selectAllRewardsByAssetId = createDeepEqualOutputSelector(
+  selectStakingDataByAccountSpecifier,
+  selectAssetId,
+  (stakingData, selectedAssetId): Record<string, chainAdapters.cosmos.Reward[]> => {
```

Comment:
```suggestion
  (stakingData, selectedAssetId): Record<CAIP19, chainAdapters.cosmos.Reward[]> => {
```

Reply:
In case we have the `ValidatorAddress` type, we could use it here instead

I have a doubt, I think it's not a CAIP19 isn't it?

Reply:
Yeay you're right this is not a CAIP19 as an indexor, the key is coming from `current.validator.address` and is effectively a cosmos account public key. We have something that works instead of `ValidatorAddress`: [`PubKey`](https://github.com/shapeshift/web/pull/1375/files/8260ecd7f43feeca69614b876b41c6d69da53dc5#diff-2c0b44901d292f739d6b8c369891eccbfb1e7a3670738db18ceb1f6f64aac98bR9). 

We can check responses in unchained as a confirmation that it is indeed a [Validator Operator bech32 account address](https://docs.cosmos.network/v0.39/basics/accounts.html) and not a CAIP19: `{REACT_APP_UNCHAINED_COSMOS_HTTP_URL}/docs/#/v1/GetAccount`

<img width="536" alt="image" src="https://user-images.githubusercontent.com/17035424/161750397-6b93c06c-a7e4-4f0f-873f-eb69d7f9d548.png">